Source/Core/Core/PowerPC/JitArm64/JitArm64_Integer.cpp

```diff
@@ -425,17 +415,17 @@ void JitArm64::cmp(UGeckoInstruction inst)
     return;
   }
 
+  gpr.BindCrToRegister(crf, false);
   ARM64Reg WA = gpr.GetReg();
   ARM64Reg WB = gpr.GetReg();
   ARM64Reg XA = EncodeRegTo64(WA);
   ARM64Reg XB = EncodeRegTo64(WB);
   ARM64Reg RA = gpr.R(a);
   ARM64Reg RB = gpr.R(b);
+
   SXTW(XA, RA);
   SXTW(XB, RB);
-
-  SUB(XA, XA, XB);
-  STR(INDEX_UNSIGNED, XA, PPC_REG, PPCSTATE_OFF(cr_val[crf]));
+  SUB(gpr.Cr(crf), XA, XB);
```

Comment:
I doubt WB is needed here. With a better emitter and SUB+SXTW in one instruction, XA isn't needed either.